### ISSUE TITLE
Json Value parsing can produce segmentation faults

### ISSUE THREAD

**Issue by MartinTheDragon** [beached/daw_json_link#398]
Trying to parse non-existing paths as strings will cause segmentation faults.

Example:

```c++
using namespace daw::json;
using std::string;

json_value json{R"({"food":"lasagna"})"};
string& favourite_food = as<string>(json["garfield"]); // SEGFAULT
```

Attempting to parse number literals in a similar manner will produce a `json_exception` with `ErrorReason::UnexpectedEndOfData` as the specified reason, which may or may not be correct.

**Comment by beached** [comment]
There was an issue here, but also, if the value doesn't exist it should throw.  `daw::json::as<std::optional<std::string>>( json )` would be correct or some other Nullable type.

**Comment by beached** [comment]
Thanks for finding, this is closed by #400
